src/txmempool.h

```diff
@@ -684,6 +684,8 @@ class CTxMemPool
         return (mapTx.count(gtxid.GetHash()) != 0);
     }
 
+    std::optional<CTxMemPoolEntry::CTxMemPoolEntryRef> GetEntry(const uint256& hash) const LIFETIMEBOUND EXCLUSIVE_LOCKS_REQUIRED(cs);
```

Comment:
Yeah, I thought this would better convey its semantics, but with all the force pushes over the past two hours, it seems like it is not intuitive to work with, so I'll revert to using a pointer.